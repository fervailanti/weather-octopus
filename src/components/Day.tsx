import React, { FC, ReactElement } from 'react'
import { DayType } from '../helpers/types'

const Day: FC<DayType> = ({ date, weather }): ReactElement => (
  <div className="flex items-center mt-4 mr-4">
    <img src={weather.icon} className="h-6 mr-2" alt="weather-icon" />
    <div>
      <p className="text-white font-thin text-sm">{date}</p>
      <p className="text-white font-thin text-sm">
        <span className="font-normal">{weather.temp.min}</span>ยบ /{' '}
        <span className="font-normal">{weather.temp.max}</span>ยบ
      </p>
    </div>
  </div>
)

export default Day
