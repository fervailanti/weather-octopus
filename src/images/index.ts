const req = (route: string) => require(`../images/gallery/${route}`)

export const images: any = {
  common: {
    octopus: req('common/octopus.png'),
    reload: req('common/reload.svg')
  },
  cities: {
    newYork: req('cities/new_york.jpg'),
    rome: req('cities/rome.jpg'),
    madrid: req('cities/madrid.jpg'),
    tokyo: req('cities/tokyo.jpg'),
    paris: req('cities/paris.jpg'),
    berlin: req('cities/berlin.jpg'),
  },
  weather: {
    icons: {
      sun: req('icons/sun.svg'),
      moon: req('icons/moon.svg'),
      sunCloudy: req('icons/sun-cloudy.svg'),
      moonCloudy: req('icons/moon-cloudy.svg'),
      clouds: req('icons/clouds.svg'),
      wind: req('icons/wind.svg'),
      rain: req('icons/rain.svg'),
      showerRain: req('icons/shower-rain.svg'),
      sunRain: req('icons/sun-rain.svg'),
      moonRain: req('icons/moon-rain.svg'),
      storm: req('icons/storm.svg'),
      snow: req('icons/snow.svg'),
      mist: req('icons/mist.svg'),
      forecast: req('icons/forecast.svg')
    }
  }
}