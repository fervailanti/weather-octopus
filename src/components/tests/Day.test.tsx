import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Day from '../Day'
import { images } from '../../images'

describe('Day Component', () => {
  it('Should render correctly', () => {
    const day = render(
      <Day
        date="Monday 20th"
        weather={{
          icon: images.weather.icons.sun,
          temp: { min: 12, max: 15 }
        }}
      />
    )
    expect(day.container).toMatchSnapshot()
  })
})
