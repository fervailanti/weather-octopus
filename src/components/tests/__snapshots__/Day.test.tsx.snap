// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day Component Should render correctly 1`] = `
<div>
  <div
    class="flex items-center mt-4 mr-4"
  >
    <img
      alt="weather-icon"
      class="h-6 mr-2"
      src="[object Object]"
    />
    <div>
      <p
        class="text-white font-thin text-sm"
      >
        Monday 20th
      </p>
      <p
        class="text-white font-thin text-sm"
      >
        <span
          class="font-normal"
        >
          12
        </span>
        ยบ /
         
        <span
          class="font-normal"
        >
          15
        </span>
        ยบ
      </p>
    </div>
  </div>
</div>
`;
